docs/source/common/lightning_module.rst

```diff
@@ -932,9 +932,93 @@ True if using TPUs
 
 --------------
 
+automatic_optimization
+~~~~~~~~~~~~~~~~~~~~~~
+When set to False, Lightning does not automate the optimization process. This means you are responsible for your own
+optimizer behavior
+
+.. code-block:: python
+
+    def __init__(self):
+        self.automatic_optimization = False
+
+    def training_step(self, batch, batch_idx):
+        # access your optimizers with use_pl_optimizer=False. Default is True
+        opt = self.optimizers(use_pl_optimizer=True)
+
+        loss = ...
+        self.manual_backward(loss, opt)
+        opt.step()
+        opt.zero_grad()
+
+This is not recommended when using a single optimizer, instead it's recommended when using 2+ optimizers
+AND you are an expert user. Most useful for research like RL, sparse coding and GAN research.
+
+In the multi-optimizer case, ignore the optimizer_idx flag and use the optimizers directly
+
+.. code-block:: python
+
+    def __init__(self):
+        self.automatic_optimization = False
+
+    def training_step(self, batch, batch_idx, optimizer_idx):
+        # access your optimizers with use_pl_optimizer=False. Default is True
+        (opt_a, opt_b) = self.optimizers(use_pl_optimizer=True)
+
+        gen_loss = ...
+        self.manual_backward(gen_loss, opt_a)
+        opt_a.step()
+        opt_a.zero_grad()
+
+        disc_loss = ...
+        self.manual_backward(disc_loss, opt_b)
+        opt_b.step()
+        opt_b.zero_grad()
```

Comment:
```suggestion
        gen_loss = ...
        opt_a.zero_grad()
        self.manual_backward(gen_loss, opt_a)
        opt_a.step()

        disc_loss = ...
        opt_b.zero_grad()
        self.manual_backward(disc_loss, opt_b)
        opt_b.step()
```

Reply:
you sure about this? I just checked a few tests related to this in there it's like:
```python
manual_backward()
step()
zero_grad()
```

Reply:
We use zero_grad after step throught Lightning, however, recently we have noticed that GANs can fail to learn if it is not done before backward. see: #5545

Reply:
yep makes sense.. since there might be some left over gradients in the generator here from `disc_loss.backward()`